pkg/volume/host_path/nsenter.go

```diff
@@ -0,0 +1,150 @@
+// +build linux
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package host_path
+
+import (
+	"fmt"
+
+	"k8s.io/utils/exec"
+)
+
+const (
+	hostProcMountsNamespace = "/rootfs/proc/1/ns/mnt"
+	nsenterCmd              = "nsenter"
+	statCmd                 = "stat"
+	touchCmd                = "touch"
+	mkdirCmd                = "mkdir"
+)
+
+// nsenterFileTypeChecker is part of experimental support for running the kubelet
+// in a container. nsenterFileTypeChecker works by executing "nsenter" to run commands in
+// the host's mount namespace.
+//
+// nsenterFileTypeChecker requires:
+//
+// 1.  The host's root filesystem must be available at "/rootfs";
+// 2.  The "nsenter" binary must be on the Kubelet process' PATH in the container's
+//     filesystem;
+// 3.  The Kubelet process must have CAP_SYS_ADMIN (required by "nsenter"); at
+//     the present, this effectively means that the kubelet is running in a
+//     privileged container;
+// 4.  The host image must have "stat", "touch", "mkdir" binaries in "/bin", "/usr/sbin", or "/usr/bin";
+
+type nsenterFileTypeChecker struct {
+	path   string
+	exists bool
+}
+
+func newNsenterFileTypeChecker(path string) (hostPathTypeChecker, error) {
```

Comment:
@thockin No tests for this now. But the interface of `newNsenterFileTypeChecker` has already been included in `host_path_test.go` as `fakeHostPathTypeChecker`. No need to test the interface again. They both have implemented interface `hostPathTypeChecker`.